internal/inventory/azurefetcher/fetcher_resource_graph.go

```diff
@@ -0,0 +1,103 @@
+	resourcesToFetch := []struct {
+		name           string
+		azureGroup     string
+		azureType      string
+		classification inventory.AssetClassification
+	}{
+		{"App Services", azurelib.AssetGroupResources, azurelib.WebsitesAssetType, inventory.AssetClassificationAzureAppService},
+		{"Container Registries", azurelib.AssetGroupResources, azurelib.ContainerRegistryAssetType, inventory.AssetClassificationAzureContainerRegistry},
+		{"Cosmos DB Accounts", azurelib.AssetGroupResources, azurelib.DocumentDBDatabaseAccountAssetType, inventory.AssetClassificationAzureCosmosDBAccount},
+		{"Cosmos DB SQL Databases", azurelib.AssetGroupResources, azurelib.CosmosDBForSQLDatabaseAssetType, inventory.AssetClassificationAzureCosmosDBSQLDatabase},
+		{"Disks", azurelib.AssetGroupResources, azurelib.DiskAssetType, inventory.AssetClassificationAzureDisk},
+		{"Elastic Pools", azurelib.AssetGroupResources, azurelib.ElasticPoolAssetType, inventory.AssetClassificationAzureElasticPool},
+		{"MySQL Flexible Servers", azurelib.AssetGroupResources, azurelib.FlexibleMySQLDBAssetType, inventory.AssetClassificationAzureSQLServer},
+		{"Resource Groups", azurelib.AssetGroupResourceContainers, azurelib.ResouceGroupAssetType, inventory.AssetClassificationAzureResourceGroup},
+		{"SQL Databases", azurelib.AssetGroupResources, azurelib.MySQLDatabaseAssetType, inventory.AssetClassificationAzureSQLDatabase},
+		{"Snapshots", azurelib.AssetGroupResources, azurelib.SnapshotAssetType, inventory.AssetClassificationAzureSnapshot},
+		{"Storage Accounts", azurelib.AssetGroupResources, azurelib.StorageAccountAssetType, inventory.AssetClassificationAzureStorageAccount},
+		{"Virtual Machines", azurelib.AssetGroupResources, azurelib.VirtualMachineAssetType, inventory.AssetClassificationAzureVirtualMachine},
+	}
+	for _, r := range resourcesToFetch {
+		f.fetch(ctx, r.name, r.azureGroup, r.azureType, r.classification, assetChan)
+	}
+}
+
+func (f *resourceGraphFetcher) fetch(ctx context.Context, resourceName, resourceGroup, resourceType string, classification inventory.AssetClassification, assetChan chan<- inventory.AssetEvent) {
+	f.logger.Infof("Fetching %s", resourceName)
+	defer f.logger.Infof("Fetching %s - Finished", resourceName)
+
+	azureAssets, err := f.provider.ListAllAssetTypesByName(ctx, resourceGroup, []string{resourceType})
+	if err != nil {
+		f.logger.Errorf("Could not fetch %s: %v", resourceName, err)
+		return
+	}
+
+	for _, item := range azureAssets {
+		name := item.Name
+		if name == "" {
+			name = item.DisplayName
+		}
+		assetChan <- inventory.NewAssetEvent(
+			classification,
+			[]string{item.Id},
+			name,
+			inventory.WithRawAsset(item),
+			inventory.WithCloud(inventory.AssetCloud{
+				Provider: inventory.AzureCloudProvider,
+				Account: inventory.AssetCloudAccount{
+					Id: item.TenantId,
+				},
+				Service: &inventory.AssetCloudService{
+					Name: "Azure",
```

Comment:
What would you recommend setting here?

Reply:
something specific to the actual resource. we do this for AWS network resources:
https://github.com/elastic/cloudbeat/blob/7b1fc44974223145210ff8c8a67870fc4c753f9a/internal/inventory/awsfetcher/fetcher_elb.go#L94-L95

i just thought we should probably do the same in azure. probably harder since we're using resource graph to fetch the resources, but maybe each resource has some metadata on it that we could use